lib/Http/Middleware/ProvisioningMiddleware.php

```diff
@@ -70,6 +70,17 @@ public function beforeController($controller, $methodName) {
 		try {
 			$this->provisioningManager->provisionSingleUser($configs, $user);
 			$password = $this->credentialStore->getLoginCredentials()->getPassword();
+
+			$mailAccount = $this->provisioningManager->getMailAccount($user);
+			$provisioningId = $mailAccount->getProvisioningId();
+			$provisioning = $this->provisioningManager->getConfigById($provisioningId);
```

Comment:
loading account and provisionings here is expensive. `$configs` holds the provisionings. could you move the logic into `$this->provisioningManager->updatePassword`? Pass the configs into the method. If the account(s) have a config with master password, set the password to the master password. else use the existing logic.

this has the same result as your current implementation, but with fewer database roundtrips

Reply:
Should be fixed in https://github.com/nextcloud/mail/pull/8917/commits/8fd4982930b1a5fc4741cdca85f89ee0c36b4e8c